src/actions/guides/http_and_routing/routing_and_params.cr

```diff
@@ -247,6 +247,33 @@ class Guides::HttpAndRouting::RoutingAndParams < GuideAction
     end
     ```
 
+    ## Routing style
+
+    By default Lucky ensures that all routes adhere to the same style. All
+    route paths are expected to use underscores unless you opt-out or change
+    the style check. You can opt-out from style checking by including
+    `Lucky::SkipRouteStyleCheck` in your action.
+
+    ```crystal
+    # src/actions/users/show.cr
+    class Users::Show < BrowserAction
+      include Lucky::SkipRouteStyleCheck
+
+      get "/users/user-:some_user_id" do
```

Comment:
I'm not sure this is a valid route 🤔  I think the variable needs to be right after the `/` like `/users/:some_user_id` otherwise it would try to match a literal `site.com/users/user-:some_user_id`. Maybe this could be an action like `Guides::GettingStarted` with the route `"/guides/getting-started"`?

Reply:
@jwoertink you're right. Just checked the `PathPart` implementation. I'll update the example accordingly.

> Maybe this could be an action like Guides::GettingStarted with the route "/guides/getting-started"?

Or should we actually support defining variables wherever in the route path?

Reply:
I don't think we should support defining variables wherever in the route path. If you wanted to prefix your values, you can define a `to_param` method on the model that does that when you pass the model to `Action.with`, I believe...

Reply:
Hmm, Rails supports this out of the box. Why wouldn't we allow specifying dynamic segments at arbitrary locations in the path?

Reply:
> Rails supports this

that's true. I guess in Rails you can do stuff like `/thing(.:format)`, but that doesn't work in Lucky.

Personally, I'm not a fan of this because I think it's confusing to read, and similar to us restricting all routes to use `_`, Lucky tries to enforce a specific coding style to keep things consistent and thus less buggy. However, maybe this is something we can open a discussion on. I'll open up the issue to start the conversation to see what others think. Maybe the benefits of doing that outweigh the potential complications? 🤔

Reply:
Sounds fair to me. 🙂 Let's revert the example here, and continue the discussion in a separate issue. 👍